ui/src/message_popup/author/questions_page.jsx

```diff
@@ -59,18 +66,37 @@ export default React.createClass({
     return questionString.split(text).length;
   },
   
+  onQuestionsReceived(err, response){
+    const allDatabaseQuestions = JSON.parse(response.text).row;
+    if(allDatabaseQuestions === undefined) {
```

Comment:
Ah got it, thanks for explaining!  Yeah the other option is to have the top-level page handle the fetching and loading.  And if it hasn't gotten back the list of questions from the server yet it can show 'Loading' or something like that and not render the `<QuestionPage />` yet.  And then when the server request comes back and it sets it in the component's state, `render` can check that and only then render the `QuestionPage`.  That way this component can require it always be passed as a prop.